mmv1/third_party/terraform/services/netapp/resource_netapp_volume_test.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    Type: MMv1     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
+package netapp_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccNetappvolume_volumeCreateExample_update(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"network_name":  acctest.BootstrapSharedServiceNetworkingConnection(t, "gcnv-network-config-1", acctest.ServiceNetworkWithParentService("netapp.servicenetworking.goog")),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetappvolume_volumeCreateExample_full(context),
+			},
+			{
+				ResourceName:            "google_netapp_volume.test_volume",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccNetappvolume_volumeCreateExample_update(context),
+			},
+			{
+				ResourceName:            "google_netapp_volume.test_volume",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccNetappvolume_volumeCreateExample_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+
+resource "google_netapp_storage_pool" "default" {
+  name = "tf-test-test-pool%{random_suffix}"
+  location = "us-central1"
+  service_level = "PREMIUM"
+  capacity_gib = "2048"
+  network = data.google_compute_network.default.id
+}
+
+
+resource "google_netapp_volume" "test_volume" {
+  location = "us-central1"
+  name = "tf-test-test-volume%{random_suffix}"
+  capacity_gib = "100"
+  share_name = "testshare"
+  storage_pool = google_netapp_storage_pool.default.name
+  protocols = ["NFSV3"]
+  smb_settings = []
```

Comment:
Can we set some SMB settings in the initial step?

Reply:
I recommend:
SHOW_SNAPSHOT, SHOW_PREVIOUS_VERSIONS
I intentionally left away a few settings which I expect the API to set automatically during volume creation, like BROWSABLE, CHANGE_NOTIFY, OPLOCKS (and maybe more or less).
TF should expect that create returns some settings which the user didn't put in.

Reply:
I thought about it again: smb_settings only apply to SMB volumes and SMb volumes cannot be created without an AD server in the test environment and setting up an AD server is pretty heavy weight and out of scope for the tests.
But please make sure a TF is prepared to receive back some default values from the create operation.

Reply:
I tested it. If smb_settings isn't specified at create, API doesn't return smb_settings on subsequent reads. That means it doesn't return any unexpected default values.